### PR TITLE
Unrevert unified adblock catalog

### PR DESCRIPTION
**Pull request by antonok-edm**
Revert of #20442

(This reverts commit 4ad53ad85e52450a7c50e6f3155461e8f90ba3e5, reversing changes made to ec922be2e00ec57a3704fd0762a846f0df41294d.)

Resolves https://github.com/brave/brave-browser/issues/33621

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc [hunk 1]
```diff
@@ -1,7 +1,7 @@
-/* Copyright (c) 2019 The Brave Authors. All rights reserved.
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at https://mozilla.org/MPL/2.0/. */
+/* Copyright (c) 2019 The Brave Software Team. Distributed under the MPL2
+ * license. This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (rillian) [inline browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc:2]: Is this new? I thought we just used "Brave Authors" following chromium.

File: browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (rillian) [inline browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc:4]: https would be better, no?

**Review by SergeyZhukovsky — APPROVED** [review]
android tests ++

**Review by rillian — APPROVED** [review]
rust++

We don't usually bump the copyright date for existing license headers, and one looks like it might be the result of a broken merge?

File: browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc (hunk 1 above)
Comment (antonok-edm) [reply browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc:2]: Not sure where these came from, tbh - I'll undo them

File: components/brave_shields/browser/ad_block_component_service_manager.cc [hunk 3]
```diff
@@ -1,9 +1,9 @@
-/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
```
Comment (antonok-edm) [inline components/brave_shields/browser/ad_block_component_service_manager.cc:1]: this is technically a "new" file, although according to GitHub it's similar enough to the previous one that it's renamed instead. I'll keep the older copyright here too.

**Review by cuba — APPROVED** [review]
Tested in iOS and everything is fine there. Code-wise too everything looks good to me.

**Review by iefremov — APPROVED** [review]
browser/net/* lgtm

**Pull request merged**